libs/vkd3d/command.c

```diff
@@ -2678,16 +2680,40 @@ struct vkd3d_queue *d3d12_device_allocate_vkd3d_queue(struct vkd3d_queue_family_
 
     queue->virtual_queue_count++;
 
+    if (command_queue)
+    {
+        vkd3d_array_reserve((void**)&queue->command_queues, &queue->command_queue_size,
+                queue->command_queue_count + 1, sizeof(*queue->command_queues));
+
+        queue->command_queues[queue->command_queue_count++] = command_queue;
+    }
+
     for (i = 0; i < queue_family->queue_count; i++)
         pthread_mutex_unlock(&queue_family->queues[i]->mutex);
 
     return queue;
 }
 
-void d3d12_device_unmap_vkd3d_queue(struct vkd3d_queue *queue)
+void d3d12_device_unmap_vkd3d_queue(struct vkd3d_queue *queue, struct d3d12_command_queue *command_queue)
 {
+    size_t i;
+
     pthread_mutex_lock(&queue->mutex);
     queue->virtual_queue_count--;
```

Comment:
Do we need both virtual_queue_count and command_queue_count?

Reply:
It might make sense for the async compute present path since that branch is creating a virtual queue without the associated d3d12 command queue, so maybe it makes sense to keep the distinction?

Reply:
We can allocate internal queues as well in which case there would be no d3d12 command queue, so unless we want to deal with random null pointers in the array this is kind of needed.

Reply:
Very well.